app/src/main/java/com/hover/stax/views/CustomTextInputLayout.java

```diff
@@ -0,0 +1,120 @@
+package com.hover.stax.views;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.content.res.TypedArray;
+import android.content.res.XmlResourceParser;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+import com.hover.stax.R;
+import com.hover.stax.utils.Utils;
+
+public class CustomTextInputLayout extends TextInputLayout {
+    private final String TAG = "CustomTextInputLayout";
+	private String hint;
+	private int inputType;
+	private TextInputLayout textInputLayout;
+	private TextInputEditText textInputEditText;
+
+	public CustomTextInputLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
+		super(context, attrs);
+
+		getAttrs(context,attrs);
+		LayoutInflater.from(context).inflate(R.layout.custom_textinputlayout, this);
+		initViews();
+		fillAttr();
+	}
+
+	//INITIALIZATIONS
```

Comment:
Why are there useless comments?

Reply:
Added these comments, because while the methods in the class are descriptive, finding certain methods amongst a bunch of others, was easier when the methods were structured in grouped purpose - > initialization, methods dealing with states, and private methods used within the class. But maybe removing comments and just add decent spacing will do.

Reply:
I think the real problem is that there are a lot more methods than there should be. Each of those methods is doing the exact same thing which means there is a way to simplify the code into just a couple methods.